managedplugin/download.go

```diff
@@ -176,6 +173,54 @@ func DownloadPluginFromHub(ctx context.Context, authToken, localPath, team, name
 	return nil
 }
 
+func downloadPluginAssetFromHub(ctx context.Context, ops HubDownloadOptions) (*cloudquery_api.PluginAsset, int, error) {
+	c, err := cloudquery_api.NewClientWithResponses(APIBaseURL(),
+		cloudquery_api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
+			if ops.AuthToken != "" {
+				req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ops.AuthToken))
+			}
+			return nil
+		}))
+	if err != nil {
+		return nil, -1, fmt.Errorf("failed to create Hub API client: %w", err)
+	}
+
+	target := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+	aj := "application/json"
+
+	switch {
+	case ops.TeamName != "":
```

Comment:
nit: Personally I would have used `if`/`else` here (or even just `if` with return), as it's a switch with a single case :)

Reply:
I [agree](https://github.com/cloudquery/plugin-pb-go/pull/162#discussion_r1391471178) but leaned towards reviewer has final say

Reply:
Any reason to prefer a switch given it is a single case + default @disq ?

Reply:
Subjectively it looks more symmetrical and clearer than an if block or an if else statement if it's the last thing a fn does.